pennylane/qnode_new/qnode.py

```diff
@@ -0,0 +1,920 @@
+        """
+
+        self._metric_tensor_subcircuits = None  #: dict[tuple[int], dict[str, Any]]: circuit descriptions for computing the metric tensor
+        # introspect the quantum function signature
+        _get_signature(self.func)
+
+        self.output_conversion = None  #: callable: for transforming the output of :meth:`.Device.execute` to QNode output
+        self.output_dim = None  #: int: dimension of the QNode output vector
+
+
+    def __repr__(self):
+        """String representation."""
+        detail = "<QNode: device='{}', func={}, wires={}>"
+        return detail.format(self.device.short_name, self.func.__name__, self.num_wires)
+
+
+    def _set_variables(self, args, kwargs):
+        """Store the current values of the free parameters in the Variable class
+        so the Operators may access them.
+
+        Args:
+            args (tuple[Any]): positional (differentiable) arguments
+            kwargs (dict[str, Any]): auxiliary arguments
+        """
+        Variable.free_param_values = np.array(list(_flatten(args)))
+        if not self.mutable:
+            # only immutable circuits access auxiliary arguments through Variables
+            Variable.kwarg_values = {k: np.array(list(_flatten(v))) for k, v in kwargs.items()}
+
+
+    def _op_descendants(self, op, only):
+        """Descendants of the given operator in the quantum circuit.
+
+        Args:
+            op (Operator): operator in the quantum circuit
+            only (str, None): the type of descendants to return.
+
+                - ``'G'``: only return non-observables (default)
+                - ``'E'``: only return observables
+                - ``None``: return all descendants
+
+        Returns:
+            list[Operator]: descendants in a topological order
+        """
+        succ = self.circuit.descendants_in_order((op,))
+        if only == 'E':
+            return list(filter(_is_observable, succ))
+        if only == 'G':
+            return list(itertools.filterfalse(_is_observable, succ))
+        return succ
+
+
+    def _append_op(self, op):
+        """Append a quantum operation into the circuit queue.
+
+        Args:
+            op (~.operation.Operator): quantum operation to be added to the circuit
+        """
+        if op.num_wires == Wires.All:
+            if set(op.wires) != set(range(self.num_wires)):
+                raise ValueError("Operator {} must act on all wires".format(op.name))
+
+        # Make sure only existing wires are used.
+        for w in op.wires:
+            if w < 0 or w >= self.num_wires:
+                raise QuantumFunctionError("Operation {} applied to invalid wire {} "
+                                           "on device with {} wires.".format(op.name, w, self.num_wires))
+
+        # observables go to their own, temporary queue
+        if isinstance(op, Observable):
+            if op.return_type is None:
+                self.queue.append(op)
```

Comment:
What about renaming `self.queue` to `self.op_queue`? I think that would be clearer as the observable queue is also called `obs_queue`.

Reply:
I'll do that later when the old QNode is replaced. Right now the new QNode has to work alongside the old, and the measurement functions expect the queue to be named self.queue.

Reply:
Done.